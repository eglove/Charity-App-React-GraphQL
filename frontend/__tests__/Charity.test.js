import CharityComponent from "../components/Charity";
import {shallow, mount} from 'enzyme';
import toJson from "enzyme-to-json";

const fakeCharity = {
    id: 'ABC123',
    ein: '123456',
    name: 'Good Dooers',
    description: 'We do good',
    website: 'goodooer.com',
    image: 'image.png',
    largeImage: 'largeImage.png',
    imageDescription: 'descriptive description',
    user: 'user1',
};

describe('<Charity/>', () => {

    it('renders and matches the snapshot', () => {
        const wrapper = shallow(<CharityComponent charity={fakeCharity}/> );
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    // it('renders the image properly', () => {
    //     const wrapper = shallow(<CharityComponent charity={fakeCharity}/>);
    //     const img = wrapper.find('img');
    //     expect(img.props().src).toBe(fakeCharity.image);
    //     expect(img.props().alt).toBe(fakeCharity.imageDescription);
    // });
    //
    // it('renders the description properly', () => {
    //     const wrapper = shallow(<CharityComponent charity={fakeCharity}/>);
    //     const p = wrapper.find('p');
    //     expect(p.text()).toBe('We do good...');
    // });
});