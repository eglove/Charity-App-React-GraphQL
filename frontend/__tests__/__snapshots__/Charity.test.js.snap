// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Charity/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <Link
    href={
      Object {
        "pathname": "/charity",
        "query": Object {
          "id": "ABC123",
        },
      }
    }
  >
    <a>
      <img
        alt="descriptive description"
        src="image.png"
      />
    </a>
  </Link>
  <p>
    We do good
    ...
  </p>
  <FavoritesButtonList
    charity={
      Object {
        "description": "We do good",
        "ein": "123456",
        "id": "ABC123",
        "image": "image.png",
        "imageDescription": "descriptive description",
        "largeImage": "largeImage.png",
        "name": "Good Dooers",
        "user": "user1",
        "website": "goodooer.com",
      }
    }
  />
</ItemStyles__Item>
`;
